hack/verify-mocks.sh

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env bash
+
+# Copyright 2021 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script checks whether updating of Mock files generated from Interfaces
+# is needed or not. We should run `hack/update-mocks.sh` 
+# if Mock files are out of date.
+# Usage: `hack/verify-mocks.sh`.
+
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+export KUBE_ROOT
+source "${KUBE_ROOT}/hack/lib/init.sh"
+
+# Explicitly opt into go modules, even though we're inside a GOPATH directory
+export GO111MODULE=on
+
+_tmp="${KUBE_ROOT}/_tmp"
+mkdir -p "${_tmp}"
+
+"${KUBE_ROOT}/hack/update-mocks.sh"
+
+# If there are untracked generated mock files which needed to be committed
+if git_status=$(git status --porcelain --untracked=no 2>/dev/null) && [[ -n "${git_status}" ]]; then
+  echo "!!! Dirty tree. Clean up and try again."
+  exit 1
+fi
```

Comment:
It's a bit hard to determine which files amongst the untracked ones are mock files. When we see untracked files, let's just show a warning and proceed with the verification for tracked files.

```suggestion
# If there are untracked generated mock files which needed to be committed
if git_status=$(git status --porcelain --untracked=yes 2>/dev/null) && [[ -n "${git_status}" ]]; then
  echo "!!! You may have untracked mock files."
fi
```

Reply:
done